tag/map.go

```diff
@@ -146,11 +146,16 @@ func Delete(k Key) Mutator {
 // NewMap returns a new tag map originated from the incoming context
 // and modified with the provided mutators.
 func NewMap(ctx context.Context, mutator ...Mutator) (*Map, error) {
-	// TODO(jbd): Implement validation of keys and values.
 	m := newMap(0)
 	orig := FromContext(ctx)
 	if orig != nil {
 		for k, v := range orig.m {
+			if !checkKeyName(k.Name()) {
```

Comment:
nit: shall we check for nil before this?

Reply:
Would be a nice check although it can't be nil though as that's a `Key{}` not `*Key{}`

Reply:
Ahh that's right, LGTM then.